src/apps/tools/OswAppStepStats.cpp

```diff
@@ -43,7 +39,10 @@ void OswAppStepStats::drawInfoPanel(OswUI* ui, uint32_t pos, uint32_t lastWeekDa
     hal->gfx()->setTextCenterAligned();
     hal->gfx()->setTextBottomAligned();
     hal->gfx()->setTextCursor(DISP_W / 2, 170);
-    hal->gfx()->print(hal->getLocalWeekday(&pos));
+    OswDate oswDate = { };
+    hal->getLocalDate(oswDate, &pos);
+    const char* weekday = oswDate.weekDayName;
+    hal->gfx()->print(weekday);
```

Comment:
why did you move this down here? I think this code was just a block of drawing instructions before, but now it also contains logic? maybe fence it of into an own `{}` section